core/templates/services/rte-helper-modal.controller.ts

```diff
@@ -412,24 +436,37 @@ export class RteHelperModalComponent {
             this.ngbActiveModal.dismiss('cancel');
           }
         );
-    } else {
+    } else if (this.componentId === this.COMPONENT_ID_VIDEO) {
       for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
         const caName = this.tmpCustomizationArgs[i].name;
         if (caName === 'video_id') {
           const temp = this.tmpCustomizationArgs[i].value;
           customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
             temp.toString()
           );
-        } else if (caName === 'text' && this.currentRteIsLinkEditor) {
-          // Set the link `text` to the link `url` if the `text` is empty.
+        } else {
           (
```

Comment:
We keep repeating this. Is it possible instead, for each component, to update this.tmpCustomizationArgs[i].value directly so that the generic logic can be done at the end?

Reply:
If we are going to keep it divided by componentId only I think it is a bit difficult. It is of course possible to use the old logic with the for loop beginning before the video, link and other rte-types and perform the same logic but maybe it messes up the division by type?

Reply:
I'm not sure I understand why it is difficult. Can you elaborate?

The idea here is to first modify this.tmpCustomizationArgs[i].value directly in each of the special cases, and then have generic logic at the end to set customizationArgsDict[caName] = this.tmpCustomizationArgs[i].value. Am I missing something here?

Reply:
@seanlip Sorry but I am not really sure what you mean by this. Since we begin with an if-clause for math, and then continue with else-if for video, link and a finishing else for the other RTE I have trouble to put the generic logic at the end. After investing further the only thing we really need to make the RTE-components other than math work is this
```ts
for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
        const caName = this.tmpCustomizationArgs[i].name;

        if (caName === 'video_id') {
          const temp = this.tmpCustomizationArgs[i].value;
          customizationArgsDict[caName] = this.extractVideoIdFromVideoUrl(
            temp.toString()
          );
        } else {
          (
            customizationArgsDict as {
              [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
                CustomizationArgsNameAndValueArray[number]['value'];
            }
          )[caName] = this.tmpCustomizationArgs[i].value;
        }
```
which basically is the code before dividing it by componentId. Maybe you can help guide me in your proposed way so I can implement it? Thanks

Reply:
Hi @yungmulah, no worries. I was wondering if you could do something like this:

```
    // First, do all the component-specific logic for each component.

    if (this.componentId === this.COMPONENT_ID_VIDEO) {
      for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
        const caName = this.tmpCustomizationArgs[i].name;
        if (caName === 'video_id') {
          const temp = this.tmpCustomizationArgs[i].value;
          this.tmpCustomizationArgs[i].value = this.extractVideoIdFromVideoUrl(
            this.tmpCustomizationArgs[i].value.toString()
          );
        }
      }
   } else if (this.componentId === ...) {
      // other component-specific stuff
   } else if (...) {
      // ...
   }
 
   // Second, do the generic logic once this.tmpCustomizationArgs values are all correct.

    for (let i = 0; i < this.tmpCustomizationArgs.length; i++) {
    const caName = this.tmpCustomizationArgs[i].name;
      (
      customizationArgsDict as {
       [Prop in CustomizationArgsNameAndValueArray[number]['name']]:
          CustomizationArgsNameAndValueArray[number]['value'];
      }
      )[caName] = this.tmpCustomizationArgs[i].value;
    }

    this.ngbActiveModal.close(customizationArgsDict);
```

Do you think an approach like this will work (note that the generic code only is done once at the end), or will it cause problems?

Reply:
Oh I see. Yea this approach seems to work fine, thanks for the guidance!